src/jobs/jobs.controller.ts

```diff
@@ -335,23 +318,27 @@ export class JobsController {
   /**
    * Checking if user is allowed to create job according to auth field of job configuration
    */
+  // TBD
   async instanceAuthorization(
-    jobCreateDto: CreateJobDto,
+    jobParams: Record<string, any> | undefined,
     user: JWTUser,
   ): Promise<boolean> {
     // Accepted options:
     // #all, #datasetOwner, #datasetOwnerOrAccess, #authenticated,
 
-    if (jobCreateDto.jobParams.auth != JobsAuth.All) {
+    if (jobParams === undefined) {
```

Comment:
`jobParams` is currently optional. If it is not provided then what would be the default user auth? The current implementation allows for job creation and update when `jobParams` is undefined.

When `jobParams` is provided, we assume that it contains both `auth` and `datasetIds` fields, otherwise we throw an exception. Is some extra validation needed or is this enough?

Reply:
It is reasonable to assume that jobParams is always present.
I thought that the validation rules for jobParams were defined in the job definition.